BUILD.bazel

```diff
@@ -0,0 +1,79 @@
+load("@rules_cc//cc:defs.bzl", "cc_library")
+
+package(default_visibility = ["//visibility:public"])
+
+constraint_setting(name = "fastfloat", default_constraint_value = ":without_fastfloat")
+constraint_value(name = "with_fastfloat", constraint_setting = ":fastfloat")
+constraint_value(name = "without_fastfloat", constraint_setting = ":fastfloat")
```

Comment:
are tests going to be built always? Ideally, they'd be optional.

Reply:
Bazel targets are only built if requested

Reply:
I should have been more clear. Here's a quick run down:

1. A bazel project only pulls external dependencies they've declared in their `WORKSPACE.bazel` file.
2. The test targets `visibility` is private. Technically they couldn't build them unless they patched the `BUILD.bazel` file or provided their own.
3. `deps` are only fetched if they are used. This is to say that every dependency is lazy loaded.

Simple example

```python
cc_library(name = "a", deps = ["@c4core"])
cc_library(name = "b", deps = ["@some_other_external_lib"])
```

When someone were to build `b` like so:

```sh
bazel build //:b
```

`c4core` wouldn't be fetched. It would only be fetched if they built `a`

```sh
//bazel build //:a
```

The same would apply to doctest. If they don't use any of the test targets (and they can't because they are private) then doctest will not be used.